catalog/ui/catalog-ui-search/src/main/java/org/codice/ddf/catalog/ui/metacard/ListApplication.java

```diff
@@ -0,0 +1,323 @@
+    if (storableResource.isError()) {
+      errorMessages.add(storableResource.getErrorMessage());
+    }
+    return storableResource;
+  }
+
+  private AttachmentInfo getAttachmentInfo(StorableResource storableResource) throws IOException {
+    return new AttachmentInfoImpl(
+        storableResource.getInputStream(),
+        storableResource.getFilename(),
+        storableResource
+            .getMimeType()
+            .orElse(contentTypeFromFilename(storableResource.getFilename())));
+  }
+
+  private AttachmentInfo parseAttachments(List<Part> contentParts) {
+
+    if (contentParts.size() == 1) {
+      Part contentPart = contentParts.get(0);
+      return parseAttachment(contentPart);
+    }
+    LOGGER.debug("There was not exactly one attachment as expected: count={}", contentParts.size());
+    return null;
+  }
+
+  private AttachmentInfo parseAttachment(Part contentPart) {
+    InputStream stream = null;
+
+    try {
+      stream = contentPart.getInputStream();
+    } catch (IOException e) {
+      LOGGER.info("IOException reading stream from file attachment in multipart body", e);
+    }
+
+    return attachmentParser.generateAttachmentInfo(
+        stream, contentPart.getContentType(), contentPart.getSubmittedFileName());
+  }
+
+  private String contentTypeFromFilename(String filename) {
+    String fileExtension = FilenameUtils.getExtension(filename);
+    String contentType = null;
+    try {
+      contentType = mimeTypeMapper.getMimeTypeForFileExtension(fileExtension);
+    } catch (MimeTypeResolutionException e) {
+      LOGGER.debug("Unable to get contentType based on filename extension {}", fileExtension);
+    }
+    LOGGER.debug("Refined contentType = {}", contentType);
+    return contentType;
+  }
+
+  private List<Splitter> lookupSplitters(String mimeType) throws MimeTypeParseException {
+    List<Splitter> splitters = splitterLocator.find(new MimeType(mimeType));
+    if (CollectionUtils.isEmpty(splitters)) {
+      LOGGER.debug("Unable to find a splitter for mime-type {}", mimeType);
+    }
+    return splitters;
+  }
+
+  private void store(
+      AttachmentInfo createInfo,
+      Consumer<String> idConsumer,
+      Consumer<String> errorMessageConsumer) {
+
+    CreateStorageRequest streamCreateRequest =
+        new CreateStorageRequestImpl(
+            Collections.singletonList(
+                new IncomingContentItem(
+                    uuidGenerator,
+                    createInfo.getStream(),
+                    createInfo.getContentType(),
+                    createInfo.getFilename())),
+            null);
+    try {
+      CreateResponse createResponse = catalogFramework.create(streamCreateRequest);
+
+      createResponse.getCreatedMetacards().stream().map(Metacard::getId).forEach(idConsumer);
+
+    } catch (IngestException e) {
+      String errorMessage = "Error while storing entry in catalog.";
+      LOGGER.info(errorMessage, e);
```

Comment:
Should we add an entry to the ingest logs as well?

Reply:
done